MovieApp/MovieApp/App/Repository/MovieRepository.swift

```diff
@@ -1,42 +1,76 @@
 import Foundation
 
 class MovieRepository: MovieRepositoryProtocol {
-    
+
     static let shared: MovieRepositoryProtocol = MovieRepository()
     
     private let networkDataSource: MovieNetworkDataSourceProtocol!
+    private let userDefaultsDataSource: UserDefaultsDataSourceProtocol!
     
+    var categoryMovies: [LocalCategory: [MovieRepoModel]]!
+        
     init() {
         self.networkDataSource = MovieNetworkDataSource.shared
+        self.userDefaultsDataSource = UserDefaultsDataSource.shared
+        
+        categoryMovies = [:]
     }
     
     func fetchPopularMovies(completion: @escaping (Result<[MovieRepoModel], RequestError>) -> Void) {
         networkDataSource.fetchPopularMovies { [weak self] result in
-            self?.mapResult(result: result, completion: completion)
+            self?.mapResult(result: result, category: .popular, completion: completion)
         }
     }
     
-    func fetchTrendingToday(completion: @escaping(Result<[MovieRepoModel], RequestError>) -> Void) {
+    func fetchTrending(completion: @escaping(Result<[MovieRepoModel], RequestError>) -> Void) {
+        var today = false
+        var week = false
         networkDataSource.fetchTrendingToday { [weak self] result in
-            self?.mapResult(result: result, completion: completion)
+            today = true
```

Comment:
Since you're using `self` more than once, please guard it.

Reply:
[Here](https://github.com/lihter/Movie-App/pull/38/commits/cd9b26a2e35cb7418ace1f07855e235f95125719#diff-119fd471623eca33515c1b8fb62109b006cbeee1e1df1dc1a8e9a2e5ba5d4c27R29)